### PR TITLE
fix: update gaec destination with config validation

### PR DESCRIPTION
**Pull request by ItsSudip**
## What are the changes introduced in this PR?

We are adding validations on configs of gaec destination.

## What is the related Linear task?

Resolves INT-XXX

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added in **new readability format**?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3847/test-report.html

File: src/v0/destinations/google_adwords_enhanced_conversions/transform.js [hunk 1]
```diff
@@ -45,11 +55,19 @@ const responseBuilder = async (metadata, message, { Config }, payload) => {
...
-    } else throw new ConfigurationError(`LoginCustomerId is required as subAccount is true.`);
+  if (subAccount) {
+    if (!loginCustomerId) {
+      throw new ConfigurationError(`LoginCustomerId is required as subAccount is true.`);
+    }
+    if (isNumber(loginCustomerId)) {
+      loginCustomerId = loginCustomerId.toString();
+    }
+    if (loginCustomerId && !isString(loginCustomerId)) {
+      throw new InstrumentationError('loginCustomerId should be a string or number');
```
Comment (utsabc) [inline src/v0/destinations/google_adwords_enhanced_conversions/transform.js:66]: Lets add a test case to cover this line

File: src/v0/destinations/google_adwords_enhanced_conversions/transform.js [hunk 2]
```diff
@@ -35,7 +36,16 @@ const updateMappingJson = (mapping) => {
 const responseBuilder = async (metadata, message, { Config }, payload) => {
   const response = defaultRequestConfig();
   const { event } = message;
-  const filteredCustomerId = removeHyphens(Config.customerId);
+  const { subAccount } = Config;
+  let { customerId, loginCustomerId } = Config;
+  if (isNumber(customerId)) {
```
Comment (krishna2020) [inline src/v0/destinations/google_adwords_enhanced_conversions/transform.js:41]: does config support numeric data types ? I'm expecting regardless on user input, customerId always gets sent as string to transformer.

Comment (ItsSudip) [reply src/v0/destinations/google_adwords_enhanced_conversions/transform.js:41]: The dynamic config allows us to expect numeric data from the payload. The bugsnag alert was caused by the dynamic config only.

File: src/v0/util/index.test.js [hunk 3]
```diff
@@ -968,3 +969,18 @@ describe('isAxiosError', () => {
...
   });
 });
+
+describe('removeHyphens', () => {
+  const data = [
+    { input: 'hello-w--orld', expected: 'helloworld' },
+    { input: '', expected: '' },
+    { input: null, expected: null },
+    { input: undefined, expected: undefined },
+    { input: 12345, expected: 12345 },
```
[the remark this answers is not in the document]
Comment (ItsSudip) [reply src/v0/util/index.test.js:979]: added

File: src/v0/destinations/google_adwords_enhanced_conversions/transform.js (hunk 1 above)
Comment (ItsSudip) [reply src/v0/destinations/google_adwords_enhanced_conversions/transform.js:66]: Added

**Review by utsabc — APPROVED** [review]

**Pull request merged**